common/util_test.go

```diff
@@ -1628,13 +1665,74 @@ func TestSecondsToDuration(t *testing.T) {
 		2: 2 * time.Second,
 	} {
 		t.Run(strconv.Itoa(int(seconds)), func(t *testing.T) {
-			got := SecondsToDuration(seconds)
+			got := common.SecondsToDuration(seconds)
 			require.Equal(t, want, got)
 		})
 	}
 }
 
 func TestNewPerTaskListScope(t *testing.T) {
-	assert.NotNil(t, NewPerTaskListScope("test-domain", "test-tasklist", types.TaskListKindNormal, metrics.NewNoopMetricsClient(), 0))
-	assert.NotNil(t, NewPerTaskListScope("test-domain", "test-tasklist", types.TaskListKindSticky, metrics.NewNoopMetricsClient(), 0))
+	assert.NotNil(t, common.NewPerTaskListScope("test-domain", "test-tasklist", types.TaskListKindNormal, metrics.NewNoopMetricsClient(), 0))
+	assert.NotNil(t, common.NewPerTaskListScope("test-domain", "test-tasklist", types.TaskListKindSticky, metrics.NewNoopMetricsClient(), 0))
+}
+
+func TestCheckEventBlobSizeLimit(t *testing.T) {
+	for name, c := range map[string]struct {
+		blobSize int
+		warnSize int
+		errSize  int
+		wantErr  error
+	}{
+		"blob size is less than limit": {
+			blobSize: 10,
+			warnSize: 20,
+			errSize:  30,
+			wantErr:  nil,
+		},
+		"blob size is greater than warn limit": {
+			blobSize: 21,
+			warnSize: 20,
+			errSize:  30,
+			wantErr:  nil,
+		},
+		"blob size is greater than error limit": {
+			blobSize: 31,
+			warnSize: 20,
+			errSize:  30,
+			wantErr:  common.ErrBlobSizeExceedsLimit,
+		},
+		"error limit is less then warn limit": {
+			blobSize: 21,
+			warnSize: 30,
+			errSize:  20,
+			wantErr:  common.ErrBlobSizeExceedsLimit,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			testScope := tally.NewTestScope("test", nil)
+			metricsClient := metrics.NewClient(testScope, metrics.History)
+			logger := testlogger.New(t)
```

Comment:
there are simpler noop versions of these, may be nicer if they're not going to be asserted on.

Reply:
Switched to log.MockLogger, though I'm confused as to why it is not generated.
I guess I'll update the mock in a separate PR. I think mock is manually generated with mockery and has a weird interface for On("Method," args...), which IMHO is even worse than what mock provides.